### DISCUSSION TITLE
lnd with custom tor domain name generated by mkp224o

### DISCUSSION THREAD

**Discussion by jrydval** [lightningnetwork/lnd#5763] [Q&A] [answered]
Hi guys! 

I generated a few vanity .onion addresses for my lnd node using mkp224o utility. 

Now I have them in hs_ed25519_secret_key files. How can I convert them for my lnd node? Can I "embed" the keys base64 encoded into v3_onion_private_key file?

Or is there a super-simple way?

Best Regards,
Jan

**Comment by guggero** [comment]
See discussion in https://github.com/lightningnetwork/lnd/issues/5750.

Just out of curiosity: where does the sudden interest in vanity tor addresses come from? You're the third person in a week with that question.

**Reply by jrydval** [reply]
It is probably coincidence :-) I connected my lightning node to the TOR network on Friday and started browsing some TOR web sites and I realised that there are vanity domains... and also started wanting one.

Thanks for the reference!

Jan

**Reply by lnd-routing** [reply]
We may have started this with our vanity address. First on the network that I'm aware of. Oops.

**Comment by xanoni** [comment, marked as answer]
I have a script to do exactly this here: https://github.com/xanoni/tor-misc/blob/main/tor-privkey-to-base64.sh
